packages/resources/src/util/apiGatewayV1AccessLog.ts

```diff
@@ -0,0 +1,133 @@
+    return;
+  }
+
+  // note: Access log configuration is not supported by L2 constructs as of CDK v1.85.0. We
+  //       need to define it at L1 construct level.
+
+  // create log group
+  let logGroup;
+  let destinationArn;
+  if (
+    accessLog &&
+    (accessLog as apig.CfnStage.AccessLogSettingProperty).destinationArn
+  ) {
+    destinationArn = (accessLog as apig.CfnStage.AccessLogSettingProperty)
+      .destinationArn;
+  } else {
+    const root = scope.node.root as App;
+    const apiName = root.logicalPrefixedName(scope.node.id);
+    // Backwards compatibility, only suffix if not default stage
+    const logGroupName =
+      "LogGroup" + (isDefaultStage ? "" : stageName);
+    let retention = logs.RetentionDays.INFINITE;
+    if (
+      accessLog &&
+      (accessLog as AccessLogProps).retention
+    ) {
+      retention = (accessLog as AccessLogProps).retention ||
+        logs.RetentionDays.INFINITE;
+    }
+    logGroup = new logs.LogGroup(scope, logGroupName, {
+      logGroupName: [
+        `/aws/vendedlogs/apis`,
+        `/${cleanupLogGroupName(apiName)}`,
+        `/${cleanupLogGroupName(stageName)}`,
+      ].join(""),
+      retention,
+    });
+    destinationArn = logGroup.logGroupArn;
+  }
+
+  // get log format
+  let format: string;
+  if (
+    accessLog &&
+    (accessLog as apig.CfnStage.AccessLogSettingProperty).format
+  ) {
+    format = (accessLog as apig.CfnStage.AccessLogSettingProperty).format as string;
+  } else if (
+    accessLog &&
+    (accessLog as AccessLogProps).format
+  ) {
+    format = (accessLog as AccessLogProps).format as string;
+  } else if (typeof accessLog === "string") {
+    format = accessLog;
+  } else {
+    format = "{" +
+        [
+          // request info
+          `"requestTime":"$context.requestTime"`,
+          `"requestId":"$context.requestId"`,
+          `"httpMethod":"$context.httpMethod"`,
+          `"path":"$context.path"`,
+          `"resourcePath":"$context.resourcePath"`,
+          `"status":$context.status`, // integer value, do not wrap in quotes
+          `"responseLatency":$context.responseLatency`, // integer value, do not wrap in quotes
+          `"xrayTraceId":"$context.xrayTraceId"`,
+          // integration info
+          `"integrationRequestId":"$context.integration.requestId"`,
+          `"functionResponseStatus":"$context.integration.status"`,
+          `"integrationLatency":"$context.integration.latency"`,
+          `"integrationServiceStatus":"$context.integration.integrationStatus"`,
+          // caller info
+          `"ip":"$context.identity.sourceIp"`,
+          `"userAgent":"$context.identity.userAgent"`,
+          `"principalId":"$context.authorizer.principalId"`,
+        ].join(",") +
+        "}";
+  }
+
+  const accessLogData = { 
+    logGroup,
+    format: apig.AccessLogFormat.custom(format),
+    destination: (new LogGroupDestination(destinationArn) as unknown) as apig.LogGroupLogDestination,
```

Comment:
We haven't been using the `bind()` idea in other places. Let's maybe keep it simple and do something like:
```js
destination: { destinationArn } as apig.LogGroupLogDestination
```

Reply:
This is what the RestApi does internally when creating a Stage (see `@aws-cdk/aws-apigateway/lib/stage.ts`):
```
accessLogSetting = {
        destinationArn: accessLogDestination?.bind(this).destinationArn,
        format: accessLogFormat?.toString() ? accessLogFormat?.toString() : AccessLogFormat.clf().toString(),
      };
```
We can't just use { destinationArn }.